### ISSUE TITLE
xeus-sqlite kernel not in the list of kernels

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/vscode-jupyter#11240]
### Discussed in https://github.com/microsoft/vscode-jupyter/discussions/11236

<div type='discussions-op-text'>

<sup>Originally posted by **bucweat** August 31, 2022</sup>
Trying to xeus-sqlite in a Jupyter notebook in Visual Studio Code so that I can execute sqlite SQL queries directly. So far I've not been able to see/select the xsqlite kernel in Visual Studio Code. I've searched google and here and, while there are lots of items here regarding loading of kernels, I've not found anything that helps (yet). 

Running Macbook Pro with latest MacOS. Conda is latest and greatest and just did the xeus-sqlite install today. I've installed using conda to a virtual environment with name 'sqlite' per the xeus-sqlite instructions at https://xeus-sqlite.readthedocs.io/en/latest/installation.html.

```
conda create -n sqlite
conda activate sqlite
conda install xeus-sqlite notebook -c conda-forge
jupyter notebook .
```

I create a new notebook, and xsqlite kernel is available and can be selected and xeus-sqlite magic and SQL commands (e.g. %LOAD magic) work as expected...

![notebook_kernels](https://user-images.githubusercontent.com/3979338/187668203-8d267454-310c-40b1-a984-e3d9efc233c3.png)

I verified that xsqlite shows up with `jupyter kernelspec list`:

<img width="576" alt="terminal_kernels" src="https://user-images.githubusercontent.com/3979338/187668534-4f544d33-3563-47a8-8205-38a0e2518c06.png">

Here is what my kernel list looks like in Visual Studio Code:

<img width="638" alt="vscode_kernels" src="https://user-images.githubusercontent.com/3979338/187668894-f47d0ce4-babd-4728-a1d4-c7574507cef9.png">

As you can see, the conda sqlite virtual env shows up, but only the python kernel is present and not the xsqlite kernel. I do see that the path to the kernel in Visual Studio Code is different from that reported by `jupyter kernelspec list` so maybe that is the issue...just not sure how to correctly configure that if so...

This is a pretty niche requirement and so not much found with google-fu. I'm hoping someone here has been down this path already...
</div>

**Comment by DonJayamanne** [comment]
@bucweat

Thanks for filing this issue, its a bug and sorry you're running into this.

* Please could you install the pre-release version of the Jupyter extension
* Open settings & go into `Jupyter->Logging`
* Change the value to `verbose`
* Reload VS Code,
* Attempt to repro the issue & then copy the output from the `Jupyter` output panel.

**Comment by DonJayamanne** [comment]
I'm on a Mac my self, however on M1, with Apple Silicon chip and xeus-sqlite isn't available for that chip.
Will try this on windows, in the mean time please send the logs.

**Comment by bucweat** [comment]
Hi and thanks for the reply.

I should note that I was running VSCodium which does not seem to have the ability to run pre-release version. I reinstalled VSCode...Microsoft happy that I'm once again on their radar...

Also note that I do have Python extension loaded in all cases. So assuming per this comment from the code that it should find non-python kernels like xsqlite: https://github.com/microsoft/vscode-jupyter/blob/589f8cbba9d97966d8d993011b42c7121e032fdd/src/kernels/raw/finder/localPythonAndRelatedNonPythonKernelSpecFinder.node.ts#L40

I do see a couple of places where the code ignores kernels where the language is not python (e.g. https://github.com/microsoft/vscode-jupyter/blob/589f8cbba9d97966d8d993011b42c7121e032fdd/src/kernels/raw/finder/localPythonAndRelatedNonPythonKernelSpecFinder.node.ts#L133) I'm wondering about this since xeus-sqlite reports language as `sqlite` at least as parsed from my .ipynb file...could this be an issue?

Anyway, log output attached. Note that the notebook I'm opening was previously opened via `jupyter notebook` server, and so it points to the `xsqlite` kernel. From the log it looks like the jupyter extension figures that out, but cannot find it in the list of available kernels. 

[jupyterlog.txt](https://github.com/microsoft/vscode-jupyter/files/9467284/jupyterlog.txt)

Hope that helps...

**Comment by DonJayamanne** [comment]
> e code ignores kernels where the language is not python (e.g.

That has nothing to do with the issue,

> og it looks like the jupyter extension figures that out, but cannot find it in the list of available kernels.

This is definitely a bug, and for some reason its not hitting an expected code path.
I'll provide a custom VSIX shortly with additional logging.

**Comment by DonJayamanne** [comment]
Please install this custom VSIX using the command `Extensions: Install from VSIX`
Repeat the same steps and please provide the logs

[ms-toolsai-jupyter-insiders.vsix.zip](https://github.com/microsoft/vscode-jupyter/files/9475417/ms-toolsai-jupyter-insiders.vsix.zip)

**Comment by bucweat** [comment]
Hi Don,

So I had to run this on insiders version as it would not load in the normal version:

<img width="267" alt="Screen Shot 2022-09-03 at 8 16 03 AM" src="https://user-images.githubusercontent.com/3979338/188255940-767ada7d-4a36-4c42-9bec-c6afe7f56006.png">

Here is my insider version info just in case:

```
Version: 1.72.0-insider (Universal)
Commit: c4722f1bb292284e0437eb7547808a5788fe6430
Date: 2022-09-02T05:20:22.688Z
Electron: 19.0.12
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin x64 21.6.0
Sandboxed: Yes
```

In terms of kernels listed, the xsqlite is not listed so everything looks the same as before...which is as expected as the new .vsix file is for additional log collection. Speaking of log, here ya go:

[jupyter log 20220903.txt](https://github.com/microsoft/vscode-jupyter/files/9481802/jupyter.log.20220903.txt)

Did a compare with the previous log there is definitely more stuff. This line indicates it is looking in the right folder for `xsqlite`:

```
debug 08:26:00.175: Searching for kernels in /Users/charlie/miniconda3/envs/sqlite/share/jupyter/kernels

```

Another note of interest: one of the other conda envs I have `jupyterlab` also has two kernels: 

```
(jupyterlab) charlie@charlieMBP ~ % jupyter kernelspec list  
Available kernels:
  python3    /Users/charlie/miniconda3/envs/jupyterlab/share/jupyter/kernels/python3
  xpython    /Users/charlie/miniconda3/envs/jupyterlab/share/jupyter/kernels/xpython
```

But maybe that one is being ignored by this in the code

https://github.com/microsoft/vscode-jupyter/blob/f49679994de5678098160682534030555e318d89/src/kernels/raw/finder/localKernelFinder.node.ts#L205

Anyway, hope something here is helpful...

**Comment by DonJayamanne** [comment]
I believe I've identified the problem:
Please could you install this VSIX, test this issue once again and upload the logs
[ms-toolsai-jupyter-insiders.vsix (1).zip](https://github.com/microsoft/vscode-jupyter/files/9486484/ms-toolsai-jupyter-insiders.vsix.1.zip)

**Comment by bucweat** [comment]
It works! Running with insider same as above. xsqlite kernel loads and is selected when I open the .ipynb file and the cells run and output as expected. 

The `CommandNotFoundError: Your shell has not been properly configured to use 'conda activate'.` around line 520 of the log file may be a byproduct of installing and then uninstalling mamba as I've not seen that before today. I did get something similar in the terminal for mamba when I ran it for the first time, but I don't get it when I run conda, so not sure what is going on there...

[20220905 jupyter log.txt](https://github.com/microsoft/vscode-jupyter/files/9486955/20220905.jupyter.log.txt)
